### ISSUE TITLE
Regex.split should not have named argument

### ISSUE THREAD

**Issue by JonathanStarup** [flix/flix#9471]
the types are not the same

**Comment by gwydd12** [comment]
Hey @JonathanStarup

First, thanks for the great work on flix.
I have some questions regarding the issue, currently it only addresses the `split` function, however, 
looking through the flix code it seems that `named parameters` are used throughout the Regex implementation even though the types of these parameter differ in some cases (e.g `getPrefix`, `getSuffix`). This observation leads me to the following question.

Would it make sense to generalize this and refactor the necessary functions as well?

If so I would love to take on this issue and work on it.

**Comment by JonathanStarup** [comment]
You are right, this applies to many functions. I think this was copied from String without this named argument consideration in mind.

**Comment by gwydd12** [comment]
Thanks for the fast response.

May I ask if the refactoring of the named parameters in the Regex implementation should be done in one PR or should it be split into groups of smaller PRs?
And of course, if it would be okay if I could try to refactor this?

**Comment by JonathanStarup** [comment]
I think it makes sense as one PR, give it a shot :) if you have questions during the process feel free to ask any time

**Comment by magnus-madsen** [comment]
Just so we are on the same page, can you give three examples of the type signatures you plan to change?

**Comment by magnus-madsen** [comment]
That sounds good. @JonathanStarup and @stephentetley are you OK with these changes? They do seem to make the API more regular.

**Comment by stephentetley** [comment]
Seems fine to me.

**Comment by magnus-madsen** [comment]
Fixed in master.
